src/components/Daypickr.js

```diff
@@ -91,12 +92,16 @@ const Daypickr = ({
   }, [isDialogOpen, dialogRef.current]);
 
   useLayoutEffect(() => {
-    if (isDialogOpen) {
-      yearSelectRef.current.focus();
-    } else if (document.activeElement === inputRef.current) {
-      return;
+    if (!firstUpdate.current) {
+      if (isDialogOpen) {
+        yearSelectRef.current.focus();
+      } else if (document.activeElement === inputRef.current) {
+        return;
+      } else {
+        toggleButtonRef.current.focus();
+      }
     } else {
-      toggleButtonRef.current.focus();
+      firstUpdate.current = false;
```

Comment:
toto mi pride kusok divne. neexistuje na to nieco krajsie? ja by som sa pytal v devdep, ale ked povies ze sa to ma robit takto, tak to tak bude :)

mozno by som to prementoval na isInitialRender

Reply:
https://stackoverflow.com/questions/53253940/make-react-useeffect-hook-not-run-on-initial-render aj novsie riesenia tam stale vyuzivaju to `useref()` max to mozem dat do custom hooku ale to mi pride ako zbytocne komplikovanie

Reply:
no a na druhom riadku riesnia pisu 

If we want the effect to run in the same phase that componentDidUpdate does, we can use [useLayoutEffect](https://reactjs.org/docs/hooks-reference.html#uselayouteffect) instead.

hlavne sa mi nepacia tie vnorene podmienky.

Reply:
hej videl som to a myslim ze to nie je pravda ... ved `useLayoutEffect` tam mame + v tom linku sa pise: If you’re migrating code from a class component, note useLayoutEffect fires in the same phase as componentDidMount and componentDidUpdate. However, we recommend starting with useEffect first and only trying useLayoutEffect if that causes a problem. Neviem ako to myslel ten komentar.